test/framework/flux.go

```diff
@@ -274,7 +346,7 @@ func (e *ClusterE2ETest) validateWorkerNodeReplicaUpdates(ctx context.Context) e
 	return e.validateWorkerNodeUpdates(ctx)
 }
 
-func (e *ClusterE2ETest) validateWorkerNodeUpdates(ctx context.Context) error {
+func (e *ClusterE2ETest) validateWorkerNodeUpdates(ctx context.Context, opts ...CommandOpt) error {
```

Comment:
TODO for the future work around validation: support for multiple worker node groups; this method currently only takes the 0 index wngc into account

Reply:
same with `validateWorkerNodeMachineSpec`

Reply:
added in https://github.com/aws/eks-anywhere/issues/396

Reply:
On that topic for future work:

I would love us to build a package that takes a kubernetes client (kubectl or go-client or an interface that abstracts them) and a cluster spec and validates it. Something like `ValidateEKSACluster(spec *cluster.Spec, client KubeClient) error`

I think we have a bunch of very useful validations sparse across different places, all with different cases. I'm sure we can reused most of them. But I believe the idea of a central place to answer the question "does my cluster conform to how I specked it" is very powerful. Moreover, it will help make a bunch of parts of our system more robust, not only the tests.

Reply:
added in https://github.com/aws/eks-anywhere/issues/396